client/keystore/src/local.rs

```diff
@@ -411,36 +416,52 @@ impl KeystoreInner {
 	}
 
 	/// Get the key phrase for a given public key and key type.
-	fn key_phrase_by_type(&self, public: &[u8], key_type: KeyTypeId) -> Result<String> {
+	fn key_phrase_by_type(&self, public: &[u8], key_type: KeyTypeId) -> Result<Option<String>> {
 		if let Some(phrase) = self.get_additional_pair(public, key_type) {
-			return Ok(phrase.clone())
+			return Ok(Some(phrase.clone()))
+		} else if self.path.is_none() {
+			return Ok(None)
 		}
 
-		let path = self.key_file_path(public, key_type).ok_or_else(|| Error::Unavailable)?;
-		let file = File::open(path)?;
+		let path = self.key_file_path(public, key_type)
+			.expect("We checked above that path exists; qed");
```

Comment:
We could have removed the `self.path.is_none()` check above and just return `Ok(None)` here if `self.key_file_path` returns `None`.

Reply:
:brain: